src/store/transactions/withdraw/withdraw.thunks.ts

```diff
@@ -217,10 +218,15 @@ function withdraw(
             }
             handleTransactionSuccess(dispatch, account.accountIndex);
           })
-          .catch((error) => {
+          .catch((error: Error) => {
             console.error(error);
             dispatch(withdrawActions.stopTransactionApproval());
-            handleTransactionFailure(dispatch, error);
+
+            if (error?.message.indexOf(alreadyDone)) {
```

Comment:
I would move this logic to the `handleTransactionFailure` function, as we already have there the new logic that @amonsosanz did. I suggest this:

```ts
function handleTransactionFailure(dispatch: AppDispatch, error: unknown) {
  const withdrawAlreadyDoneErrorCode = "WITHDRAW_ALREADY_DONE";
  const errorMsg = persistence.getErrorMessage(error);
  const snackbarMsg = errorMsg.includes(withdrawAlreadyDoneErrorCode)
    ? "The withdraw has already been done"
    : errorMsg;

  dispatch(openSnackbar(`Transaction failed - ${snackbarMsg}`, theme.palette.red.main));
}
```

Doing this we take advantage of the `persistence.getErrorMessage` function and we display a better message to the end user instead just `WITHDRAW_ALREADY_DONE`, which it's quite strange for an end user.